pkg/controller/clusterautoscaler/clusterautoscaler_controller.go

```diff
@@ -289,6 +293,13 @@ func (r *Reconciler) UpdateAutoscaler(ca *autoscalingv1.ClusterAutoscaler) error
 		return err
 	}
 
+	// Update the cluster provider type.
+	platformType, err := r.getPlatformType()
+	if err != nil {
+		return err
+	}
+	r.config.platformType = platformType
```

Comment:
do we need to set this on a `CreateAutoscaler` as well, or will we get an update after creating?